test/test_pickle.py

```diff
@@ -0,0 +1,103 @@
+__copyright__ = """
+Copyright (C) 2023 University of Illinois Board of Trustees
+"""
+
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import os
+import sys
+from typing import Any, Callable, Dict, Optional
+
+import pytest
+
+from orderedsets import FrozenOrderedSet
+
+# {{{ test infrastructure
+
+
+def run_test_with_new_python_invocation(f: Callable[..., Any], *args: Any,
+                                        extra_env_vars:
+                                        Optional[Dict[str, Any]] = None) -> None:
+    if extra_env_vars is None:
+        extra_env_vars = {}
+
+    from base64 import b64encode
+    from pickle import dumps
+    from subprocess import check_call
+
+    env_vars = {
+        "INVOCATION_INFO": b64encode(dumps((f, args))).decode(),
+    }
+    env_vars.update(extra_env_vars)
+
+    my_env = os.environ.copy()
+    my_env.update(env_vars)
+
+    check_call([sys.executable, __file__], env=my_env)
+
+
+def run_test_with_new_python_invocation_inner() -> None:
+    from base64 import b64decode
+    from pickle import loads
+    f, args = loads(b64decode(os.environ["INVOCATION_INFO"].encode()))
+
+    f(*args)
+
+# }}}
+
+
+# {{{ test that pickling a FrozenOrderedSet recomputes the hash on unpickling
+
+@pytest.mark.xfail(reason="needs fix for caching hash value issue")
+def test_pickle_hash() -> None:
+    from pickle import dumps
+
+    f1 = FrozenOrderedSet(["a", "b", "c"])
+    print(hash(f1))  # Force creating a cached hash value
+
+    assert f1._my_hash
+    run_test_with_new_python_invocation(_test_pickle_hash_stage2, dumps(f1))
```

Comment:
> In addition, you could supply a `PYTHONHASHSEED`, 

Wouldn't the hash change even when not supplying `PYTHONHASHSEED`? (or is the goal to prevent that `PYTHONHASHSEED` specified by the user interferes with the test?)

> pass in the hash, and assert non-equality. To be fair, that's not guaranteed, but highly probable.

Thanks, added in #33.

Reply:
> Wouldn't the hash change even when not supplying PYTHONHASHSEED? 

Probably? :shrug: